.github/workflows/mobilecoin-dev-cd.yaml

```diff
@@ -0,0 +1,563 @@
+# Copyright (c) 2018-2022 The MobileCoin Foundation
+#
+# MobileCoin Core projects - Build, deploy to development.
+
+name: mobilecoin-dev-cd
+
+env:
+  DOCKER_ORG: mobilecoin
+  PREVIOUS_RELEASE: 1.1.3-dev
+
+on:
+  push:
+    branches:
+    # TODO: Remove after current 1.2.0 branch
+    - release-1.2.0
+    - release/v[0-9]+.[0-9]+.[0-9]+
+    - feature/*
+    - develop
+
+jobs:
+############################################
+# Generate data and environment information
+############################################
+  generate-metadata:
+    name: 👾 Environment Info 👾
+    runs-on: [self-hosted, Linux, small]
+    outputs:
+      branch: ${{ steps.meta.outputs.branch }}
+      namespace: ${{ steps.meta.outputs.namespace }}
+      tag: ${{ steps.meta.outputs.tag }}
+      docker_org: ${{ env.DOCKER_ORG }}
+      previous_release: ${{ env.PREVIOUS_RELEASE }}
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Generate version metadata
+      id: meta
+      run: |
+        .internal-ci/util/metadata.sh
+
+    - name: 👾 Print Environment Details 👾
+      env:
+        NAMESPACE: ${{ steps.meta.outputs.namespace }}
+        VERSION: ${{ steps.meta.outputs.tag }}
+      run: |
+        .internal-ci/util/print_details.sh
+
+    - name: Cache wallet seeds
+      id: wallet_seeds
+      uses: actions/cache@v3
+      with:
+        path: |
+          .tmp/wallet_seeds
+        key: ${{ github.ref_name }}-wallet-seeds
+
+    - name: Create wallet seeds
+      run: |
+        .internal-ci/util/generate_wallet_seeds.sh
+
+#########################################
+# Build binaries
+#########################################
+  build-cargo-hardware:
+    runs-on: [self-hosted, Linux, large]
+    container:
+      image: mobilecoin/rust-sgx-base@sha256:565e0b31ba40379098de222d63f13fee9fc3f7b72318743794f103f79e85fbca
+    steps:
+    - name: Checkout
+      uses: actions/checkout@v3
+
+    - name: Cache rust build binaries
+      if: "! contains(github.event.head_commit.message, '[skip build]')"
```

Comment:
If you skip a job, you then need to do a bunch of complicated if statements on each downstream job to see if the `needs` job skipped or failed.  And even then its super messy because you need to check the status and add needs for all of the upstream jobs not just the ones the current job is dependent on. 

Believe it or not I think this is way cleaner.